### PR TITLE
Correctly link to Dag parsing context in docs

### PR DESCRIPTION
**Pull request by ashb**
Before:

image

After:

image

### CODE REVIEW

**Comment by ashb** [comment]
I can split out the second commit in to a separate PR if desired.

File: docs/conf.py [hunk 1]
```diff
@@ -198,44 +194,52 @@
...
+            exclude_patterns.append(f"_api/airflow/{path.name}")
+
+    # Don't include all of utils, just the specific ones we include in python-api-ref
+    for path in (root / "utils").iterdir():
+        if path.name not in browseable_utils:
+            exclude_patterns.append(_get_rst_filepath_from_path(path))
 else:
     exclude_patterns.extend(
-        _get_rst_filepath_from_path(f) for f in glob.glob(f"{PACKAGE_DIR}/**/example_dags/**/*.py")
+        _get_rst_filepath_from_path(f) for f in pathlib.Path(PACKAGE_DIR).glob("**/example_dags")
```
Comment (ashb) [inline docs/conf.py:242]: Rather than excluding every file underneath, just exclude the whole folder.

File: docs/conf.py [hunk 2]
```diff
@@ -698,6 +701,7 @@ def _get_params(root_schema: dict, prefix: str = "", default_section: str = "")
     autoapi_ignore.append('*/airflow/providers/*')
 else:
     autoapi_ignore.append('*/airflow/providers/cncf/kubernetes/backcompat/*')
+    autoapi_ignore.append('example_dags/*')
```
Comment (ashb) [inline docs/conf.py]: Example dags were moved to point at github (and excluded in output), but we were still parsing them for autoapi

**Comment by potiuk** [comment]
OH. Bad me. :(

**Comment by potiuk** [comment]
Thanks @ashb for fixing those !

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Docs build passed. Merging.

**Pull request merged**